application/models/UserGroup.php

```diff
@@ -246,7 +246,16 @@ public function getbuttons()
 
         return $button;
     }
-
+    /**
+     * This function search usergroups for a user
+     * If $mine = true then usergroups are those that have been created by the current user
+     * else this function provides usergroups which contain the current user
+     * 
+     * The object \CActiveDataProvider returned is used to generate the view in application/views/admin/usergroup/usergroups_view.php
+     * 
+     * @param bool $mine
+     * @return \CActiveDataProvider
+     */
     function searchMine($mine)
```

Comment:
Thanks for your update. I'm going to be a bit petty and ask you to rename your `$mine` variable. Here are a couple of examples of how a boolean variable could be named to make clear its function:

```

 isFish = true
 isCat = false
 hasScales = true
 hasFur = false
 canSwim = true
 wasEgg = true
 eatAble = true
```

Google for more info about best practice.